external/llvm-project/mlir/lib/Dialect/AMDGPU/Transforms/EmulateAtomics.cpp

```diff
@@ -86,6 +89,23 @@ static void patchOperandSegmentSizes(ArrayRef<NamedAttribute> attrs,
   }
 }
 
+// A helper function to flatten a vector element into a integer equivalent
+static Value flattenVecToInt(ConversionPatternRewriter &rewriter, Location loc,
+                             Value val) {
+  if (auto vectorType = val.getType().dyn_cast_or_null<VectorType>()) {
+    int64_t bitwidth =
+        vectorType.getElementTypeBitWidth() * vectorType.getNumElements();
+    Type elemType = rewriter.getIntegerType(bitwidth);
+    SmallVector<int64_t, 2> flatVecTypeShape(1, vectorType.getRank());
+    auto flatVecType = VectorType::get(flatVecTypeShape, elemType);
+    auto bitcast = rewriter.create<vector::BitCastOp>(loc, flatVecType, val);
+    auto zero = rewriter.createOrFold<arith::ConstantIndexOp>(loc, 0);
+    auto ret = rewriter.create<vector::ExtractElementOp>(loc, bitcast, zero);
```

Comment:
I think `ExtractElementOp` is semi-deprecated, but that's not for this critical fix